functions.inc.php

```diff
@@ -1918,4 +1918,21 @@ function getRemoteAddr() {
     return $REMOTE_ADDR;
 }
 
+
+/**
+ * Returns a hash for a username valid for one day
+ *
+ * @param String $username user name
+ * @return String password recovery code
+ */
+function getPasswordRecoveryCode($username)
+{
+    $username = trim(strtolower($username));
+    $date = date('Y-m-d');
+
+    $code = substr(strtoupper(md5('SECRET SALTING PHRASE' . $username . $date)), 0, 6);
```

Comment:
The implementation is now using the existing generate_password() function to generate a random 12 alphanum characters reset token unique to the user, which is stored in database, is valid for 1 hour and can only be used once.